### PR TITLE
JAMES-2909 Use Architecture Decision Record

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: src/adr/references/documenting-architecture-decisions.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# [Documenting Architecture Decisions](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
+
+## Context
+
+Architecture for agile projects has to be described and defined
```
Comment (chibenwa) [inline src/adr/references/documenting-architecture-decisions.md]: Maybe we can say that its also true for community projects...

File: src/adr/references/documenting-architecture-decisions.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+**Context** This section describes the forces at play, including
+technological, political, social, and project local. These forces are
+probably in tension, and should be called out as such. The language in
+this section is value-neutral. It is simply describing facts.
+
+**Decision** This section describes our response to these forces. It is
+stated in full sentences, with active voice. "We will ..."
+
+**Status** A decision may be "proposed" if the project stakeholders
```
Comment (chibenwa) [inline src/adr/references/documenting-architecture-decisions.md]: Who os the project stackholder?

File: src/adr/references/documenting-architecture-decisions.md (hunk 1 above)
Comment (blackheaven) [reply src/adr/references/documenting-architecture-decisions.md]: that's a copy and paste of the original article

Comment (chibenwa) [reply src/adr/references/documenting-architecture-decisions.md]: Dont we need to adapt it?

Comment (blackheaven) [reply src/adr/references/documenting-architecture-decisions.md]: if we want to keep it as a reference, no.

File: src/adr/references/documenting-architecture-decisions.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# [Documenting Architecture Decisions](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
```
Comment (rouazana) [inline src/adr/references/documenting-architecture-decisions.md]: Licence? Or maybe we can simply link to archive.org if we are afraid of this article to be lost? https://web.archive.org/web/20190824074401/http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions

**Comment by blackheaven** [comment]
> I still miss something explaining me how to write an ADR

it's in the associated link.

I can add the formalism if you want, but a complete how to would be overkill IMO

File: src/adr/0001-record-architecture-decisions.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+In order to be more community-oriented, we should adopt a process to have a structured way to have open architectural decisions.
+
+Using an Architectural Decision Records-based process as a support of discussion on the developers mailing-lists.
+
+## Decision
+
+We will use Architecture Decision Records, as [described by Michael Nygard](https://web.archive.org/web/20190824074401/http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions).
+
+Each ADR will be discussed on the Apache James' developers mailing-list before being accepted.
```
Comment (chibenwa) [inline src/adr/0001-record-architecture-decisions.md]: Can we state:

```
Following https://community.apache.org/committers/decisionMaking.html , we provide the following possible status, with their associated meaning:

 - `Proposed`: The decision is being discussed on the mailing list.
 - `Accepted (lazy consensus)` : the architecture decision was proposed on the mailing list, and a consensus emerged from people involved in the discussion on the mailing list.
 - `Accepted (voted)` : the architecture undergo a voting process.
 - `Rejected` : Consensus builded up against that proposal.
```

Also:

```
We should provide in a mutable `Reference` section links to related JIRA meta-ticket (not necessarily to all related sub  tickets) as well as a link to the mail archive discussion thread.

JIRA tickets implementing that architecture decision should also link the related Architecture Decision Record.
```

Comment (blackheaven) [reply src/adr/0001-record-architecture-decisions.md]: We lack also `Conditionally accepted`, `Accepted and withdrawed`, etc.

I'm in favor of being a bit more fluid about that.

Comment (rouazana) [reply src/adr/0001-record-architecture-decisions.md]: I add the possible status

**Comment by rouazana** [comment]
see  #169.

**Pull request closed without merging**